### ISSUE TITLE
Deno panics unexpectedly when put under heavy load

### ISSUE THREAD

**Issue by umgefahren** [denoland/deno#15124]
We are currently developing a server benchmark for various languages and runtimes. While developing our stress-tester we discovered a bug in Deno.

This is the backtrace:
```
Platform: macos x86_64
Version: 1.23.3
Args: ["deno", "run", "--allow-net", "mod.ts"]

thread 'main' panicked at 'internal error: entered unreachable code: Unexpected missing emit: file://.../server-language-benchmark/Deno/src/server.ts

Try reloading with the --reload CLI flag or deleting your DENO_DIR.', cli/proc_state.rs:571:23
stack backtrace:
   0:        0x11005690b - <unknown>
   1:        0x10f6ab36a - <unknown>
   2:        0x1100397f8 - <unknown>
   3:        0x110045d1d - <unknown>
   4:        0x110045a21 - <unknown>
   5:        0x10f63b72b - <unknown>
   6:        0x1100467b9 - <unknown>
   7:        0x110056da4 - <unknown>
   8:        0x110056d19 - <unknown>
   9:        0x110046145 - <unknown>
  10:        0x1112ffc43 - <unknown>
  11:        0x10f4ee8d3 - <unknown>
  12:        0x10f4eed07 - <unknown>
  13:        0x10f7925b8 - <unknown>
  14:        0x10f7c6399 - <unknown>
  15:        0x110463538 - <unknown>
  16:        0x1104630b2 - <unknown>
  17:        0x110462557 - <unknown>
```

The code we ran can be found here: https://github.com/umgefahren/server-language-benchmark/tree/main/Deno

The code of the client can be found here: https://github.com/umgefahren/server-language-client

Steps to reproduce the issue with the client:
- Build the client (in release mode) `cargo build --release`
- Generate data with `./target/release/server-language-client generate 10000000`
- Run the benchmark (after starting the deno server) `./target/release/server-language-client benchmark 1m`

The issue might be somewhere hidden in the TCP stack, although I'm definitely not certain. But we did some unholy things with TCP in order to put such a high load on the server (https://github.com/umgefahren/server-language-client/blob/bc5eaf085f399cb76a4aa1e796696c5fd274c360/src/worker.rs#L99).

I consider this bug somewhat dangerous, as it might increase vulnerability to DOS attacks.

**Comment by umgefahren** [comment]
Additional Note: I only tested on the same computer, **not** over network.

**Comment by bartlomieju** [comment]
CC @dsherret

**Comment by dsherret** [comment]
It's probably #13302. I am actually working on a fix right now for 1.24.
